crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +375,122 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    /// Returns the oldest settlement event for which the accociated auction is
+    /// not yet populated in the database.
+    pub async fn get_settlement_without_auction(
+        &self,
+    ) -> Result<Option<domain::settlement::Event>, error::SettlementEvent> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_settlement_without_auction"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await.map_err(DatabaseError)?;
```

Comment:
nit: Needless mapping to an unrelated error type which later requires you to implement `impl From<DatabaseError> for error::SettlementEvent` for no reason. Either make `DatabaseError(#[from] sqlx::Error)` and simply `?` here or

```suggestion
        let mut ex = self.postgres.pool.begin().await.map_err(error::SettlementEvent::DatabaseError)?;
```

Reply:
refactored